hack/cluster-clean.sh

```diff
@@ -120,22 +120,25 @@ function delete_resources() {
 function delete_namespaces() {
     local managed_namespaces=("$@")
 
-    for i in ${managed_namespaces[@]}; do
-        if [ -n "$(_kubectl get ns | grep "${i} ")" ]; then
-            echo "Clean ${i} namespace"
-            _kubectl delete ns ${i}
-
-            local start_time=0
-            local sample=10
-            local timeout=120
-            echo "Waiting for ${i} namespace to disappear ..."
-            while [ -n "$(_kubectl get ns | grep "${i} ")" ]; do
+    for ns in ${managed_namespaces[@]}; do
+        if [ -n "$(_kubectl get ns | grep "${ns} ")" ]; then
+            echo "Clean ${ns} namespace"
+            _kubectl delete ns ${ns}
+
+            local current_time=0
+            local sample=3
+            local timeout=180
+            echo "Waiting for ${ns} namespace to disappear ..."
+            set +x
+            while [ -n "$(_kubectl get ns | grep -w ${ns})" ]; do
                 sleep $sample
-                start_time=$((current_time + sample))
+                local current_time=$((current_time + sample))
```

Comment:
I thik the `local` keyword is superfluous here, isn't it?

Reply:
else the var is accessible out of the loop

Reply:
It's already local though
```
function delete_namespaces() {
  local current_time=12
  current_time=42
  time=24
  local time=334
}

delete_namespaces

echo "current_time: $current_time"
echo "time: $time"
```
yields
```
 rhrazdil@rhrazdilt480s  ~  bash test.sh
current_time: 
time: 24
```

Reply:
ah you right forgot it was declared before